### PR TITLE
Add starter pack tests to master build

### LINKED ISSUE #1682: Add Starter pack builds to master branch

### PR DESCRIPTION
**Pull request by MetcalfeTom**
To fix #1682 

**Proposed changes**:
- use regex to parse new version builds
- add `branch = 'master'` as a condition

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] updated the changelog

### CODE REVIEW

**Comment by MetcalfeTom** [comment]
@wochinge the regex conditional build for new version branches was tested using [travis-conditions](https://github.com/travis-ci/travis-conditions)

File: .travis.yml [hunk 1]
```diff
@@ -72,15 +72,15 @@ jobs:
     - export ${!TRAVIS*}
     - sphinx-versioning push docs newdocs . -- -b dirhtml  -A html_theme=rasabaster
   - stage: test
-    if: branch = "*.x" # only new NLU version builds test the starter packs
+    if: branch =~ /(\d\.\d{1,2}\.x)/ or branch = "master" # only new version PRs & PRs to master will test starter packs
```
[the remark this answers is not in the document]
Comment (MetcalfeTom) [reply .travis.yml]:
```suggestion
    if: branch =~ /(\d+\.\d+\.x)/ or branch = "master" # only new version PRs & PRs to master will test starter packs
```

[the remark this answers is not in the document]
Comment (MetcalfeTom) [reply .travis.yml]: The lowercase `or` works as well 👍

Here's to rasa_core 10.0 🚀

[the remark this answers is not in the document]
Comment (wochinge) [reply .travis.yml]: nice 👍 😆

**Pull request merged**